### DISCUSSION TITLE
When build 64 bit Apk,lost my ane in the Apk?

### DISCUSSION THREAD

**Discussion by Ekitty** [airsdk/Adobe-Runtime-Support#1505] [General]
I'm using IntelliJ ,using the added -arch armv8 launch parameter to build , I see APK include lib/arm64-v8a/libc++_shared.so libCore.so libysshared.so.

     When I debug my App ,  I get the following error:

     [Fault] exception, information=ReferenceError: Error #1065: Variable com.freshplanet.ane.AirVideo::AirVideo is not defined.

	 I'm using AIR 33.1.1.713

	Using the added -arch  armv7 , Ane is working.

**Comment by ajwfrost** [comment]
Hi
Does the FreshPlanet ANE support an armv8 target? (if you unzip it, you should see the list of Android platforms that it supports).
Also can you check whether the various Android folders contain .so files anywhere? or is it purely .jar files?
thanks

**Comment by Ekitty** [comment]
Hi
  I unzip FreshPlanet ANE， Android-ARM folder  is  purely .jar files.  Don't .so files.

<img width="856" alt="image" src="https://user-images.githubusercontent.com/859912/147725278-e4c6a38d-3791-432c-9ce4-26c0d547cae8.png">

  ```
 extension.xml  file:
     <extension xmlns="http://ns.adobe.com/air/extension/15.0">
  <id>com.freshplanet.AirVideo</id>
  <versionNumber>1.0</versionNumber>
  <platforms>
    <platform name="iPhone-ARM">
            <applicationDeployment>
                <nativeLibrary>libAirVideo.a</nativeLibrary>
                <initializer>AirVideoInitializer</initializer> 
                <finalizer>AirVideoFinalizer</finalizer>
            </applicationDeployment>
    </platform>
 
   <platform name="Android-ARM"> 
        <applicationDeployment> 
            <nativeLibrary>libAirVideo.jar</nativeLibrary> 
            <initializer>com.freshplanet.ane.AirVideo.Extension</initializer> 
            <finalizer>com.freshplanet.ane.AirVideo.Extension</finalizer> 
        </applicationDeployment>
    </platform>
    
    <platform name="default"> 
        <applicationDeployment/> 
    </platform> 
  </platforms>
</extension>

```
    makefile :

> Package AirVideo.swc ANE
> ==========================================================
> /Applications/Adobe\ Flash\ Builder\ 4.7/eclipse/plugins/com.adobe.flash.compiler_4.7.0.349722/AIRSDK/bin/adt -package  -storetype pkcs12 -keystore a.p12 -storepass *** -tsa none -target ane /Users/AirVideo.ane extension.xml  -swc AirVideo.swc -platform Android-ARM  -C platform/android .   -platform iPhone-ARM  -C platform/ios . -platform default -C platform/default library.swf
> ==========================================================
> AirVideo.ane  is complete.


      Question:Does the FreshPlanet ANE support an armv8 target?

**Reply by ajwfrost** [reply]
So yes, the ANE doesn't support Androdi-ARM64, but you could add it by updating the extension.xml file (duplicate the `Android-ARM` section and rename it to `Android-ARM64`) and by adding `-platform Android-ARM64 -C platform/android .` to the package command...

**Reply by Policy56** [reply]
It works ! Thanks @ajwfrost  👑 !

**Comment by Ekitty** [comment]
OK！Thanks。😊
